src/App.tsx

```diff
@@ -47,49 +59,78 @@ export function getReorderedGoods(
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+export class App extends React.Component<{}, ReorderOptions> {
+  state: Readonly<ReorderOptions> = {
+    sortType: SortType.NONE,
+    isReversed: false,
+  };
 
-      <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
-      </ul>
-    </div>
-  );
-};
+  handleSortByAlpabet = () => this.setState({ sortType: SortType.ALPABET });
+
+  handleSortByLength = () => this.setState({ sortType: SortType.LENGTH });
+
+  handleReverse = () => (this.setState(state => ({
+    isReversed: !state.isReversed,
+  })));
+
+  handleReset = () => (this.setState({
+    sortType: SortType.NONE,
+    isReversed: false,
+  }));
+
+  render() {
+    const { sortType, isReversed } = this.state;
+    const onReorder = (sortType !== SortType.NONE || isReversed);
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={classNames(
+              'button is-info',
+              { 'is-light': sortType !== SortType.ALPABET },
+            )}
+            onClick={this.handleSortByAlpabet}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={classNames(
+              'button is-success',
+              { 'is-light': sortType !== SortType.LENGTH },
+            )}
+            onClick={this.handleSortByLength}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={classNames(
+              'button is-warning',
+              { 'is-light': !isReversed },
+            )}
+            onClick={this.handleReverse}
+          >
+            Reverse
+          </button>
+
+          {onReorder && (
+            <button
+              type="button"
+              className="button is-danger is-light"
+              onClick={this.handleReset}
+            >
+              Reset
+            </button>
+          )}
+        </div>
+
+        <ListOfGoods goods={getReorderedGoods(goodsFromServer, this.state)} />
```

Comment:
it will be better to save result of this function to variable and just pass it here than pass this function call